In [1]:
import pandas as pd
pd.__version__

u'0.20.2'

In [2]:
# Chapter 5
props = pd.read_csv('../data/committees.csv')
props.head()

ocd_prop_id  calaccess_prop_id  \
0  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
1  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
2  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
3  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
4  ocd-contest/85990193-9d6f-4600-b8e7-bf1317841d82            1362198   

   ccdc_prop_id                                          prop_name  \
0            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
1            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
2            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
3            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
4            71  PROPOSITION 052 - STATE FEES ON HOSPITALS. FED...   

   ccdc_committee_id  calaccess_committee_id  \
0                382                 1374469   
1                383                 1220380   
2                384                 1282321   
3                385                 1382843   
4                386                 1362973   

                                      committee_name committee_position  
0  YES ON PROPOSITION 51 - CALIFORNIANS FOR QUALI...            SUPPORT  
1  COMMUNITY COLLEGE FACILITY COALITION ISSUES CO...            SUPPORT  
2  TORLAKSON'S INVEST IN CALIFORNIA A BALLOT MEAS...            SUPPORT  
3  CALIFORNIA TAXPAYERS AND EDUCATORS OPPOSED TO ...             OPPOSE  
4  YES ON PROPOSITION 52 - A COALITION OF CALIFOR...            SUPPORT

In [3]:
contribs = pd.read_csv('../data/contributions.csv')
contribs.head()

calaccess_committee_id                              committee_name  \
0                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
1                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
2                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
3                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
4                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   

   calaccess_filing_id date_received contributor_lastname  \
0              2083796    2016-09-18              BERGMAN   
1              2083796    2016-09-18                KAHLE   
2              2083796    2016-07-15             MCDEVITT   
3              2083796    2016-08-10      WARREN-SAMARIPA   
4              2083796    2016-07-26      WARREN-SAMARIPA   

  contributor_firstname contributor_city contributor_state contributor_zip  \
0              GRETCHEN    SPRING VALLEY                CA           91978   
1                 MYRNA        SAN DIEGO                CA           92109   
2                   LEO        ESCONDIDO                CA           92025   
3             STEPHANIE        SAN DIEGO                CA           92116   
4             STEPHANIE        SAN DIEGO                CA           92116   

        contributor_employer contributor_occupation  \
0                 A NEW PATH     EXECUTIVE DIRECTOR   
1   NATIONAL SCHOOL DISTRICT                TEACHER   
2              LIFE IONIZERS    SEO/CONTENT MANAGER   
3  STEPHANIE WARREN SAMARIPA           ENTREPRENEUR   
4  STEPHANIE WARREN SAMARIPA           ENTREPRENEUR   

   contributor_is_self_employed  amount  
0                         False    84.0  
1                         False    35.0  
2                         False   198.0  
3                         False   -50.0  
4                          True    50.0

In [4]:
props.prop_name.value_counts()

PROPOSITION 057 - CRIMINAL SENTENCES. JUVENILE CRIMINAL PROCEEDINGS AND SENTENCING. INITIATIVE CONSTITUTIONAL AMENDMENT AND STATUTE.                           13
PROPOSITION 056 - CIGARETTE TAX TO FUND HEALTHCARE, TOBACCO USE PREVENTION, RESEARCH, AND LAW ENFORCEMENT. INITIATIVE CONSTITUTIONAL AMENDMENT AND STATUTE.    12
PROPOSITION 064- MARIJUANA LEGALIZATION. INITIATIVE STATUTE.                                                                                                   11
PROPOSITION 066- DEATH PENALTY. PROCEDURES. INITIATIVE STATUTE.                                                                                                 9
PROPOSITION 055 - TAX EXTENSION TO FUND EDUCATION AND HEALTHCARE. INITIATIVE CONSTITUTIONAL AMENDMENT.                                                          8
PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON SINGLE-USE PLASTIC BAGS.                                                                                         7
PROPOSITION 062- DEATH PENAL

In [5]:
prop_67 = props[props.prop_name == 'PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON SINGLE-USE PLASTIC BAGS.']
prop_67

ocd_prop_id  calaccess_prop_id  \
95   ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
96   ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
97   ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
98   ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
99   ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
100  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
101  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   

     ccdc_prop_id                                          prop_name  \
95             86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
96             86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
97             86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
98             86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
99             86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
100            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
101            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   

     ccdc_committee_id  calaccess_committee_id  \
95                 477                 1372900   
96                 478                 1372902   
97                 479                 1329332   
98                 480                 1374885   
99                 481                 1346973   
100                482                 1381938   
101                483                 1389817   

                                        committee_name committee_position  
95   YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT  
96   AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE  
97   SAVE THE BAG BAN, YES ON 67, SPONSORED BY ENVI...            SUPPORT  
98   YES ON 67 - CALIFORNIANS AGAINST WASTE - PROTE...            SUPPORT  
99   CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...            SUPPORT  
100      SAVE THE BAY ACTION FUND PAC - YES ON PROP 67            SUPPORT  
101  SAVE THE BAY ACTION FUND COMMITTEE TO SUPPORT ...            SUPPORT

In [6]:
prop_67.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7 entries, 95 to 101
Data columns (total 8 columns):
ocd_prop_id               7 non-null object
calaccess_prop_id         7 non-null int64
ccdc_prop_id              7 non-null int64
prop_name                 7 non-null object
ccdc_committee_id         7 non-null int64
calaccess_committee_id    7 non-null int64
committee_name            7 non-null object
committee_position        7 non-null object
dtypes: int64(4), object(4)
memory usage: 504.0+ bytes


In [7]:
contribs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56379 entries, 0 to 56378
Data columns (total 13 columns):
calaccess_committee_id          56379 non-null int64
committee_name                  56379 non-null object
calaccess_filing_id             56379 non-null int64
date_received                   56379 non-null object
contributor_lastname            56375 non-null object
contributor_firstname           53411 non-null object
contributor_city                56369 non-null object
contributor_state               56363 non-null object
contributor_zip                 56366 non-null object
contributor_employer            48572 non-null object
contributor_occupation          53273 non-null object
contributor_is_self_employed    56379 non-null bool
amount                          56379 non-null float64
dtypes: bool(1), float64(1), int64(2), object(9)
memory usage: 5.2+ MB


In [8]:
# merge the contributions and propostions dataset relevant to prop_67
merged = pd.merge(prop_67, contribs, on='calaccess_committee_id')
merged.head()

ocd_prop_id  calaccess_prop_id  \
0  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
1  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
2  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
3  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
4  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   

   ccdc_prop_id                                          prop_name  \
0            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
1            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
2            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
3            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
4            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   

   ccdc_committee_id  calaccess_committee_id  \
0                477                 1372900   
1                477                 1372900   
2                477                 1372900   
3                477                 1372900   
4                477                 1372900   

                                    committee_name_x committee_position  \
0  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
1  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
2  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
3  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
4  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   

                                    committee_name_y  calaccess_filing_id  \
0  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              1962528   
1  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              1982014   
2  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              1982014   
3  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              1997827   
4  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              2038229   

  date_received                               contributor_lastname  \
0    2015-02-24  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...   
1    2015-05-06  CALIFORNIA LEAGUE OF CONSERVATION VOTERS ISSUE...   
2    2015-04-21                  NATURAL RESOURCES DEFENSE COUNCIL   
3    2015-09-28  RETAILERS FOR A BETTER CALIFORNIA SPONSORED BY...   
4    2016-01-19                                      SUPER A FOODS   

  contributor_firstname  contributor_city contributor_state contributor_zip  \
0                   NaN        SACRAMENTO                CA           95814   
1                   NaN           OAKLAND                CA           94612   
2                   NaN          NEW YORK                NY           10011   
3                   NaN        SACRAMENTO                CA           95814   
4                   NaN  CITY OF COMMERCE                CA           90040   

  contributor_employer contributor_occupation  contributor_is_self_employed  \
0                  NaN                    NaN                         False   
1                  NaN                    NaN                         False   
2                  NaN                    NaN                         False   
3                  NaN                    NaN                         False   
4                  NaN                    NaN                         False   

     amount  
0  100000.0  
1    5000.0  
2    5000.0  
3    5000.0  
4    1000.0

In [9]:
merged.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6279 entries, 0 to 6278
Data columns (total 20 columns):
ocd_prop_id                     6279 non-null object
calaccess_prop_id               6279 non-null int64
ccdc_prop_id                    6279 non-null int64
prop_name                       6279 non-null object
ccdc_committee_id               6279 non-null int64
calaccess_committee_id          6279 non-null int64
committee_name_x                6279 non-null object
committee_position              6279 non-null object
committee_name_y                6279 non-null object
calaccess_filing_id             6279 non-null int64
date_received                   6279 non-null object
contributor_lastname            6279 non-null object
contributor_firstname           6211 non-null object
contributor_city                6279 non-null object
contributor_state               6278 non-null object
contributor_zip                 6278 non-null object
contributor_employer            6092 non-null obje

In [10]:
# total $ spent on campaigns pro and con
merged.amount.sum()

4470574.0

In [11]:
# Which side got more large donations?
merged.committee_position.value_counts()

SUPPORT    6252
OPPOSE       27
Name: committee_position, dtype: int64

In [12]:
support = merged[merged.committee_position == 'SUPPORT']
support.amount.sum()

1789160.75

In [13]:
oppose = merged[merged.committee_position == 'OPPOSE']
oppose.amount.sum()

2681413.25

In [14]:
support_vs_oppose = support.amount.sum() / merged.amount.sum()
support_vs_oppose

0.40020828421585236

In [15]:
# what were the ten biggest contributors?
top_10_support = support.sort_values('amount', ascending=False)
top_10_support.head(10)

ocd_prop_id  calaccess_prop_id  \
5     ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
0     ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
59    ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
61    ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
6254  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
6255  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
9     ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
39    ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
6269  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
123   ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   

      ccdc_prop_id                                          prop_name  \
5               86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
0               86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
59              86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
61              86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
6254            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
6255            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
9               86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
39              86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
6269            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
123             86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   

      ccdc_committee_id  calaccess_committee_id  \
5                   477                 1372900   
0                   477                 1372900   
59                  477                 1372900   
61                  477                 1372900   
6254                481                 1346973   
6255                481                 1346973   
9                   477                 1372900   
39                  477                 1372900   
6269                483                 1389817   
123                 477                 1372900   

                                       committee_name_x committee_position  \
5     YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
0     YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
59    YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
61    YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
6254  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...            SUPPORT   
6255  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...            SUPPORT   
9     YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
39    YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
6269  SAVE THE BAY ACTION FUND COMMITTEE TO SUPPORT ...            SUPPORT   
123   YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   

                                       committee_name_y  calaccess_filing_id  \
5     YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              2038229   
0     YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              1962528   
59    YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              2098468   
61    YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              2098468   
6254  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...              1965934   
6255  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...              1965934   
9     YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              2038229   
39    YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              2083454   
6269  SAVE THE BAY ACTION FUND COMMITTEE TO SUPPORT ...              2130495   
123   YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...              2129546   

     date_received                               contributor_lastname  \
5       2016-02-

In [16]:
top_10_oppose = oppose.sort_values('amount', ascending=False)
top_10_oppose.head(10)

ocd_prop_id  calaccess_prop_id  \
141  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
132  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
140  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
142  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
133  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
134  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
150  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
148  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
144  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
135  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   

     ccdc_prop_id                                          prop_name  \
141            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
132            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
140            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
142            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
133            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
134            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
150            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
148            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
144            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
135            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   

     ccdc_committee_id  calaccess_committee_id  \
141                478                 1372902   
132                478                 1372902   
140                478                 1372902   
142                478                 1372902   
133                478                 1372902   
134                478                 1372902   
150                478                 1372902   
148                478                 1372902   
144                478                 1372902   
135                478                 1372902   

                                      committee_name_x committee_position  \
141  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
132  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
140  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
142  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
133  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
134  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
150  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
148  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
144  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
135  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   

                                      committee_name_y  calaccess_filing_id  \
141  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2038213   
132  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2012817   
140  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2038213   
142  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2038213   
133  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2012817   
134  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2012817   
150  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2083549   
148  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2064842   
144  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2064842   
135  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2012817   

    date_received                 contributor_lastname contributor_firstname  \
141    2016-01-25  FORMOSA PLASTICS CORPORATION U.S.A.         

In [17]:
# group the two committee_name_ columns as they are just duplicates due to previous merge
merged.groupby('committee_name_x').amount.sum().reset_index().sort_values('amount', ascending=False)

committee_name_x      amount
0  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...  2681413.25
5  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...   861295.75
2  SAVE THE BAG BAN, YES ON 67, SPONSORED BY ENVI...   609365.00
1  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...   200000.00
3  SAVE THE BAY ACTION FUND COMMITTEE TO SUPPORT ...    97050.00
4  YES ON 67 - CALIFORNIANS AGAINST WASTE - PROTE...    21450.00

In [18]:
# Finding the top contributors
merged.groupby(["contributor_firstname", "contributor_lastname", "committee_position"]).amount.sum().reset_index().sort_values("amount", ascending=False)

contributor_firstname contributor_lastname committee_position    amount
1019                 JULIE              PACKARD            SUPPORT  105000.0
330                 CLAIRE                PERRY            SUPPORT  100000.0
1912              THOMAS F               STEYER            SUPPORT   50000.0
102                ANNE G.              EARHART            SUPPORT   25000.0
1466                 NANCY              BURNETT            SUPPORT   19000.0
955                   JOHN               WEEDEN            SUPPORT    5000.0
1635               RICHARD               KOPCHO            SUPPORT    5000.0
976                 JOSEPH            MATTEUCCI            SUPPORT    3060.0
272                CHARLES                JONAS            SUPPORT    2850.0
804                  JANET               CARTER            SUPPORT    2600.0
92                    ANNA                GETTY            SUPPORT    2500.0
1311                  MARK               MCGINN            SUPPORT    2400.0
997                 JUDITH                 RUIZ            SUPPORT    2210.0
1023                JUNEKO               STEELE            SUPPORT    2200.0
1758                 SCOTT                ELROD            SUPPORT    2000.0
1701                  ROSS          KONINGSTEIN            SUPPORT    2000.0
1905                THOMAS                MADER            SUPPORT    2000.0
1048                  KATE                PURDY            SUPPORT    2000.0
2013                  YVAN                 FREY            SUPPORT    2000.0
382                    DAN               MORASH            SUPPORT    2000.0
481                 DENISE                 OWEN            SUPPORT    2000.0
1606               RAYMOND              PAULSON            SUPPORT    1980.0
1607               REBECCA             BENIARES            SUPPORT    1920.0
1298              MARJORIE            BERTOLINO            SUPPORT    1800.0
1302                  MARK               BUTLER            SUPPORT    1800.0
1296               MARISSA                BRAND            SUPPORT    1800.0
716                   GWEN                MINOR            SUPPORT    1800.0
738                 HERMAN                  CHU            SUPPORT    1760.0
721                   HANS               LARSEN            SUPPORT    1690.0
762                   JADE               GRAHAM            SUPPORT    1680.0
...                    ...                  ...                ...       ...
606                  ERICH                BROWN            SUPPORT      20.0
1859               SUZANNE                 KING            SUPPORT      20.0
851               JENNIFER            CHAMBLISS            SUPPORT      20.0
791                    JAN      MCFARLAND-BROWN            SUPPORT      20.0
1049                  KATE               SEFTON            SUPPORT      20.0
631                FRANCES               KAPLAN            SUPPORT      20.0
1917                   TIM             REYNOLDS            SUPPORT      20.0
241               CAROLINE              BUGANAN            SUPPORT      20.0
694                GIN IVY              SEVILLA            SUPPORT      20.0
693     GILLIAN ALEXANDRIA                CLARK            SUPPORT      20.0
662                   GAIL         WINTERBOURNE            SUPPORT      20.0
184                BRENDAN                SEELY            SUPPORT      20.0
1082                  KATY                GRANT            SUPPORT      20.0
1223               LORETTA                 COLE            SUPPORT      20.0
1771                 SHAKA              SICKELS            SUPPORT      20.0
1207                  LISA              MERCIER            SUPPORT      20.0
1196         LINDA VALLEJO             DILLAWAY            SUPPORT      20.0
1173                  LEAH            HENRIKSON            SUPPORT      20.0
1169              LAWRENCE              DILLARD            SUPPORT      20.0
470                DEBORAH             VERSEPUT            SUPPORT      20.0
4

## Extra questions to answer:

1. What percentage of donations came from people who live outside of California?
2. What are the top employers of donors who gave for and against the measure?
3. Which committees had the fewest donors?

In [19]:
# Q1 - What percentage of donations came from people who live outside of California?
outside = merged[merged.contributor_state != 'CA']
inside = merged[merged.contributor_state == 'CA']

total = merged.amount.sum()
o = outside.amount.sum()
i = inside.amount.sum()

pct = total / 100
[o / pct, i / pct]

[65.767779484245196, 34.232220515754804]

__Answer #1__

Donations from Californian donors (34.23%) were largely outweight by non Californians (65.76%) 2 to 1

In [20]:
# Q2 - What are the top employers of donors who gave for and against the measure?

oppose_employers = oppose.contributor_employer.value_counts()
support_employers = support.contributor_employer.value_counts()

[oppose_employers.head(), support_employers.head()]

[Series([], Name: contributor_employer, dtype: int64),
 RETIRED                902
 UNEMPLOYED             850
 NONE                    85
 GOOGLE                  40
 STANFORD UNIVERSITY     34
 Name: contributor_employer, dtype: int64]

__ Answer #2__

The top five employers for those __opposing__ _Proposition 64_ are:
- BOSTON SCIENTIFIC      5
- RETIRED                4
- STATE OF CALIFORNIA    4
- SELF                   3
- CITY OF SUNNYVALE      2

The top five employers for those __supporting__ _Proposition 64_ are:

- NONE                                       92
- LOS ANGELES DEPARTMENT OF WATER & POWER     7
- RETIRED                                     6
- APPLE, INC.                                 6
- SOUTHERN CALIFORNIA EDISON                  5

In [21]:
# Q3 - Which committees had the fewest donors?

merged.committee_name_x.value_counts().reset_index()

index  committee_name_x
0  SAVE THE BAG BAN, YES ON 67, SPONSORED BY ENVI...              6054
1  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...               128
2  YES ON 67 - CALIFORNIANS AGAINST WASTE - PROTE...                45
3  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...                27
4  SAVE THE BAY ACTION FUND COMMITTEE TO SUPPORT ...                23
5  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...                 2

__Answer #3__

The following committees had the fewest donnors for _Proposition 64_

- FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))
- DRUG POLICY ACTION - NON PROFIT 501C4, YES ON
- SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT
- MOMS STRONG AGAINST PROP 64
- CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY

In [23]:
merged.groupby(["contributor_lastname","committee_position"]).amount.sum().reset_index().sort_values("amount", ascending=False).head()

contributor_lastname committee_position     amount
713                    HILEX POLY CO. LLC             OPPOSE  874769.00
522   FORMOSA PLASTICS CORPORATION U.S.A.             OPPOSE  748441.50
1534                       SUPERBAG CORP.             OPPOSE  609369.75
12                  ADVANCE POLYBAG, INC.             OPPOSE  446833.00
256        CALIFORNIA GROCERS ASSOCIATION            SUPPORT  200000.00